### ISSUE TITLE
`vue/no-unused-emit-declarations` reporting with `defineEmits()` and `defineModel()` compiler marcos

### ISSUE THREAD

**Issue by n0099** [vuejs/eslint-plugin-vue#2351]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.56.0
- **eslint-plugin-vue version:** 9.19.2
- **Node version:** 21.4.0
- **Operating System:** Windows 8.1

**Please show your full configuration:**

https://github.com/n0099/open-tbm/blob/1403ed2ffc75f200426def765ecb66142ed6c085/fe/.eslintrc.cjs

**What did you do?**	

https://github.com/n0099/open-tbm/blob/1403ed2ffc75f200426def765ecb66142ed6c085/fe/src/components/Post/queryForm/widgets/InputNumericParam.vue#L22
```js
defineEmits({
    'update:modelValue': (p: KnownNumericParams) =>
        _.isString(p.name) && _.isString(p.value)
        && numericParamSubParamRangeValues.includes(p.subParam.range)
});
const modelValue = defineModel<KnownNumericParams>({ required: true });
```

**What did you expect to happen?**
Not reporting since mutate the ref `modelValue` will emit the event `update:modelValue` under the hood.

**What actually happened?**

```
`update:modelValue` is defined as emit but never used.(vue/no-unused-emit-declarations)
```

**Repository to reproduce this issue**

https://github.com/n0099/open-tbm/blob/1403ed2ffc75f200426def765ecb66142ed6c085/fe

**Comment by FloEdelmann** [comment]
Thanks for your report! `defineModel` still seems to be experimental: https://blog.vuejs.org/posts/vue-3-3#definemodel

So I think it's best to not support it in `vue/no-unused-emit-declarations` yet, since the implementation might have to change when it's released as stable, or removed again. So the workaround for now would be to disable the rule with an inline comment.

But if you fancy, you can implement support for `defineModel` in `vue/no-unused-emit-declarations` and open a PR, I'll review it.

**Comment by n0099** [comment]
It's stable now: https://blog.vuejs.org/posts/vue-3-4#definemodel-is-now-stable

**Comment by FloEdelmann** [comment]
This is being tracked in #2130 now.
